deepmd/model/multi.py

```diff
@@ -0,0 +1,366 @@
+                                                              natoms,
+                                                              input_dict,
+                                                              reuse=reuse,
+                                                              suffix='_{}'.format(fitting_key) + suffix)
+                framesize = nout if "global" in tensor_name else \
+                    natomsel[fitting_key] * nout[fitting_key]
+                output = tf.reshape(output, [-1, framesize],
+                                    name='o_{}_{}'.format(tensor_name, fitting_key) + suffix)
+
+                model_dict[fitting_key] = {}
+                model_dict[fitting_key][tensor_name] = output
+
+                if "global" not in tensor_name:
+                    gname = "global_" + tensor_name
+                    atom_out = tf.reshape(output, [-1, natomsel[fitting_key], nout[fitting_key]])
+                    global_out = tf.reduce_sum(atom_out, axis=1)
+                    global_out = tf.reshape(global_out, [-1, nout[fitting_key]],
+                                            name="o_{}_{}".format(gname, fitting_key) + suffix)
+
+                    out_cpnts = tf.split(atom_out, nout[fitting_key], axis=-1)
+                    force_cpnts = []
+                    virial_cpnts = []
+                    atom_virial_cpnts = []
+
+                    for out_i in out_cpnts:
+                        force_i, virial_i, atom_virial_i \
+                            = self.descrpt.prod_force_virial(out_i, natoms)
+                        force_cpnts.append(tf.reshape(force_i, [-1, 3 * natoms[1]]))
+                        virial_cpnts.append(tf.reshape(virial_i, [-1, 9]))
+                        atom_virial_cpnts.append(tf.reshape(atom_virial_i, [-1, 9 * natoms[1]]))
+
+                    # [nframe x nout x (natom x 3)]
+                    force = tf.concat(force_cpnts, axis=1,
+                                      name="o_force_{}".format(fitting_key) + suffix)
+                    # [nframe x nout x 9]
+                    virial = tf.concat(virial_cpnts, axis=1,
+                                       name="o_virial_{}".format(fitting_key) + suffix)
+                    # [nframe x nout x (natom x 9)]
+                    atom_virial = tf.concat(atom_virial_cpnts, axis=1,
+                                            name="o_atom_virial_{}".format(fitting_key) + suffix)
+
+                    model_dict[fitting_key][gname] = global_out
+                    model_dict[fitting_key]["force"] = force
+                    model_dict[fitting_key]["virial"] = virial
+                    model_dict[fitting_key]["atom_virial"] = atom_virial
+
+        return model_dict
+
+    def _import_graph_def_from_frz_model(self, frz_model, feed_dict, return_elements):
+        graph, graph_def = load_graph_def(frz_model)
+        return tf.import_graph_def(graph_def, input_map=feed_dict, return_elements=return_elements, name="")
```

Comment:
I don't understand how it works considering the model containing multiple outputs cannot be frozen.

Reply:
Typo.